include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,476 @@
+                    std::input_iterator_tag>>>;
+    using value_type = zip::tuple_or_pair<std::ranges::range_value_t<view_helper::maybe_const<Const, Views>>...>;
+    using difference_type = std::common_type_t<
+                                std::ranges::range_difference_t<view_helper::maybe_const<Const, Views>>...>;
+
+    iterator() = default;
+    constexpr iterator(iterator<!Const> i)
+        requires Const && (std::convertible_to<std::ranges::iterator_t<Views>,
+                                               std::ranges::iterator_t<view_helper::maybe_const<Const, Views>>> && ...)
+    : current_(std::move(i.current))
+    {}
+
+    constexpr auto operator*() const
+    {
+        return zip::tuple_transform([] (auto & i) -> decltype(auto) { return *i; }, current_);
+    }
+    constexpr iterator& operator++()
+    {
+        zip::tuple_for_each([] (auto & i) { ++i; }, current_);
+        return *this;
+    }
+    constexpr void operator++(int)
+    {
+        ++*this;
+    }
+    constexpr iterator operator++(int)
+        requires zip::all_forward<Const, Views...>
+    {
+        auto tmp = *this;
+        ++*this;
+        return tmp;
+    }
+
+    constexpr iterator& operator--()
+        requires zip::all_bidirectional<Const, Views...>
+    {
+        zip::tuple_for_each([] (auto & i) { --i; }, current_);
+        return *this;
+    }
+    constexpr iterator operator--(int)
+        requires zip::all_bidirectional<Const, Views...>
+    {
+        auto tmp = *this;
+        --*this;
+        return tmp;
+    }
+
+    constexpr iterator& operator+=(difference_type x)
+        requires zip::all_random_access<Const, Views...>
+    {
+        zip::tuple_for_each([&] <typename I> (I & i) { i += std::iter_difference_t<I>(x); }, current_);
+        return *this;
+    }
+    constexpr iterator& operator-=(difference_type x)
+        requires zip::all_random_access<Const, Views...>
+    {
+        zip::tuple_for_each([&] <typename I> (I & i) { i -= std::iter_difference_t<I>(x); }, current_);
+        return *this;
+    }
+
+    constexpr auto operator[](difference_type n) const
+        requires zip::all_random_access<Const, Views...>
+    {
+        return zip::tuple_transform([&] <typename I> (I & i) -> decltype(auto)
+        {
+            return i[std::iter_difference_t<I>(n)];
+        }, current_);
+    }
+
+    friend constexpr bool operator==(iterator const & x, iterator const & y)
+        requires (std::equality_comparable<std::ranges::iterator_t<view_helper::maybe_const<Const, Views>>> && ...)
+    {
+        if constexpr (zip::all_bidirectional<Const, Views...>)
+        {
+            return x.current_ == y.current_;
+        }
+        else
+        {
+            return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+            {
+                return ((std::get<N>(x.current_) == std::get<N>(y.current_)) || ...);
+            } (std::index_sequence_for<Views...>{});
+        }
+    }
```

Comment:
Should we also add a `operator !=`?

Reply:
I didn't even notice :D

Do you want me to add it?
The standard doesn't add it because it's auto-generated from `operator==`: https://en.cppreference.com/w/cpp/types/type_info/operator_cmp (since C++20)

Reply:
If it is auto generated, it is not needed :+1: